tests/test_preview.py

```diff
@@ -199,3 +200,66 @@ def test_preview_global_shape(
         image_key=image_key,
     )
     assert preview.global_shape == previewed_shape
+
+
+def test_preview_offset(
+    standard_data_path: str,
+    standard_image_key_path: str,
+):
+    IN_FILE_PATH = Path(__file__).parent / "test_data/tomo_standard.nxs"
+    f = h5py.File(IN_FILE_PATH, "r")
+    dataset = f[standard_data_path]
+    image_key = f[standard_image_key_path]
+
+    param_value = PreviewParam(
+        angles=None,
+        detector_y={"start": 20, "start_offset": -21, "stop": 120, "stop_offset": 21},
+        detector_x={
+            "start": 0,
+            "start_offset": 10,
+            "stop": dataset.shape[2],
+            "stop_offset": -10,
+        },
+    )
+    preview_config = parse_preview(param_value=param_value, data_shape=dataset.shape)
+
+    preview = Preview(
+        preview_config=preview_config,
+        dataset=dataset,
+        image_key=image_key,
+    )
```

Comment:
ok, I can exclude, though it doesn't really hurt to test? What if `preview` fails for some reason that our `preview_config` is weird?

Reply:
> ok, I can exclude, though it doesn't really hurt to test?

The new functionality being added is purely for `parse_preview()`, and it influences the `PreviewConfig` object generated, that's all. Creating a `Preview` from a `PreviewConfig` has nothing to do with what `parse_preview()` is doing, and a _unit test_ for `parse_preview()` shouldn't really be doing anything outside of what `parse_preview()` does.

If the test is testing both:
- `parse_preview()`
- and a `Preview` object created from the output of `parse_preview()`

and then the test fails, it becomes more complicated to figure out what caused the test to fail:
- is it `parse_preview()` that's incorrect
- or is it the creation of `Preview` that's incorrect

Whereas if the test only involves what it needs to involve (namely, `parse_preview()` and the value it returns), then the only option is that something must be wrong with `parse_preview()`.

> What if `preview` fails for some reason that our `preview_config` is weird?

There are bounds checking tests for the creation a `Preview` object, to catch incorrect `PreviewConfig` inputs: https://github.com/DiamondLightSource/httomo/blob/b39d74dd0d0e6ed59cfe53c6bc965b23f7ab57ae/tests/test_preview.py#L12-L104

If creation of a `Preview` fails due to a weird `PreviewConfig` value, then that should be tested separately from testing `parse_preview()`.

Put another way. "weird" `PreviewConfig` objects causing errors creating a `Preview` object should be tested separately to testing `parse_preview()`. Namely, in _unit tests_ for creating a `Preview` object, not in unit tests for `parse_preview()`.

Reply:
And if `parse_preview()` creates `PreviewConfig` objects with weird indices, that's best caught by only running `parse_preview()` and checking its return value, rather than needing to create a `Preview` object from the returned `PreviewConfig` to become aware of any weird indices.

Reply:
understood, I removed the irrelevant bits